### ISSUE TITLE
`broken LLVM module found: Operand is null` when using `switch` in `inline for`

### ISSUE THREAD

**Issue by leroycep** [ziglang/zig#8858]
```zig
const std = @import("std");
const io = std.io;
const meta = std.meta;
const builtin = @import("builtin");

const Hello = extern struct {
    b: Lang,
    c: Lang,
};

const Lang = enum(u16) {
    en,
    ru,
};

pub fn main() void {
    const a: [100]u8 = undefined;
    var source = io.StreamSource{ .const_buffer = io.fixedBufferStream(&a) };
    const hello = readStructForeign(source.reader(), Hello);

    std.log.info("hello = {}", .{hello});
}

pub fn readStructForeign(reader: io.StreamSource.Reader, comptime T: type) !T {
    comptime std.debug.assert(@typeInfo(T).Struct.layout != std.builtin.TypeInfo.ContainerLayout.Auto);

    var result: T = undefined;

    inline for (meta.fields(T)) |entry| {
        switch (@typeInfo(entry.field_type)) {
            .ComptimeInt, .Int => {
                @field(result, entry.name) = try reader.readIntForeign(entry.field_type);
            },
            .Struct => {
                @field(result, entry.name) = try readStructForeign(reader, entry.field_type);
            },
            .Enum => {
                @field(result, entry.name) = try reader.readEnum(entry.field_type, switch (builtin.cpu.arch.endian()) {
                    .Little => std.builtin.Endian.Big,
                    .Big => std.builtin.Endian.Little,
                });
            },
            else => {
                std.debug.panic("Add support for type {} in readStructForeign", .{@typeName(entry.field_type)});
            },
        }
    }

    return result;
}
```

Trying to run this code results in a compiler bug:

```
$ zig run zig-broken-llvm.zig
broken LLVM module found: Operand is null
  switch i1 %31, label %SwitchElse2 [
    i1 true, <null operand!>
    i1 false, <null operand!>
  ], !dbg !18218

This is a bug in the Zig compiler.thread 351862 panic:
Unable to dump stack trace: debug info stripped
fish: Job 1, 'zig run zig-broken-llvm.zig' terminated by signal SIGABRT (Abort)
```

Removing the second field from `Hello` allows it to compile and run fine.

**Comment by gballet** [comment]
I ran into  the same problem on FreeBSD-12.0, just on the call to `builtin.cpu.arch.endian()`:

```

**Comment by Vexu** [comment]
Fixed by 4618c41fa6ca70f06c7e65762d2f38d57b00818c
